src/openssl/ssl/context.cr

```diff
@@ -1,22 +1,40 @@
 class OpenSSL::SSL::Context
-  def self.default : self
-    @@default ||= new
-  end
-
   def initialize
     @handle = LibSSL.ssl_ctx_new(LibSSL.sslv23_method)
+    self.verify_mode = OpenSSL::SSL::VerifyMode::PEER
```

Comment:
I do would like very much to centralize it, it's a good default to have it enabled. So first of all we should test with #2671 whether it's actually an issue at all.

If it is, how about a `Context.new_for_server` / `Context.new_for_client`?